skopt/optimizer/base.py

```diff
@@ -220,33 +303,42 @@ def base_minimize(func, dimensions, base_estimator,
     # setting the scope for these variables
     result = None
 
-    # User suggested points at which to evaluate the objective first
-    if x0 and y0 is None:
-        y0 = list(map(func, x0))
-        n_calls -= len(y0)
+    # TODO: I tried to clean this up a little, but it is VERY confusing.
+    # Pass user suggested initialisation points to the optimizer.
+    if x0 is not None:
+        # If objective values are not given for these points, calculate them.
+        if y0 is None:
+            # Calculate the objective function for each point in x0.
+            y0 = list(map(func, x0))
+
+            # Decrease the required number of function calls.
+            n_calls -= len(y0)
 
-    # Pass user suggested initialisation points to the optimizer
-    if x0:
         if not (isinstance(y0, Iterable) or isinstance(y0, numbers.Number)):
             raise ValueError(
                 "`y0` should be an iterable or a scalar, got %s" % type(y0))
 
         if len(x0) != len(y0):
             raise ValueError("`x0` and `y0` should have the same length")
 
-
         result = optimizer.tell(x0, y0)
         result.specs = specs
 
         if eval_callbacks(callbacks, result):
             return result
 
-    # Bayesian optimization loop
+    # Bayesian optimization loop.
     for n in range(n_calls):
+        # Get next point in the search-space.
```

Comment:
The next three comments restate what the line of code says so I would remove them

Reply:
The code is easy to understand for you because you look at it every day. For newcomers the comments help and they don't do any harm to experts. But I'm not going to argue with you. I am just informing you that it would be easier for others to help you, if your code was better documented.